test/unittest-poolboy_templating.py

```diff
@@ -390,5 +390,14 @@ def test_26(self):
             recursive_process_template_strings(template, 'jinja2', template_vars), {"a": "A"}
         )
 
+    def test_27(self):
+        template = "{{ l | merge_list_of_dicts | object }}"
+        template_vars = {
+            "l": [{"a": "A", "b": "X"}, {"b": "B", "c": "C"}, {"d": "D"}]
```

Comment:
edge cases? empty dicts, but also what's the behavior in case of non-dict values? `None`,  `false`, `[]`, etc.

Reply:
Added handling for None values. Other non-dictionary values can throw an error.